src/frequenz/sdk/actor/power_distributing/_battery_status.py

```diff
@@ -223,8 +224,80 @@ async def stop(self) -> None:
         """Stop tracking battery status."""
         await cancel_and_await(self._task)
 
-        if self._select is not None:
-            await self._select.stop()
+    def _handle_status_battery(
+        self, bat_data: BatteryData
+    ) -> None:  # Any is probably not the best
+        self._battery.last_msg_correct = (
+            self._is_message_reliable(bat_data)
+            and self._is_battery_state_correct(bat_data)
+            and self._no_critical_error(bat_data)
+            and self._is_capacity_present(bat_data)
+        )
+        self._battery.last_msg_timestamp = bat_data.timestamp
+        self._battery.data_recv_timer.reset()
+
+    def _handle_status_inverter(
+        self, inv_data: InverterData
+    ) -> None:  # Any is probably not the best
+        self._inverter.last_msg_correct = (
+            self._is_message_reliable(inv_data)
+            and self._is_inverter_state_correct(inv_data)
+            and self._no_critical_error(inv_data)
+        )
+        self._inverter.last_msg_timestamp = inv_data.timestamp
+        self._inverter.data_recv_timer.reset()
+
+    def _handle_status_set_power_result(
+        self, result: SetPowerResult
+    ) -> None:  # Any is probably not the best
+        if self.battery_id in result.succeed:
+            self._blocking_status.unblock()
+
+        elif (
+            self.battery_id in result.failed and self._last_status != Status.NOT_WORKING
+        ):
+            duration = self._blocking_status.block()
+
+            if duration > 0:
+                _logger.warning(
+                    "battery %d failed last response. block it for %f sec",
+                    self.battery_id,
+                    duration,
+                )
+
+    def _handle_status_battery_timer(
+        self, _timedelta: timedelta
+    ) -> None:  # Any is probably not the best
+        if self._battery.last_msg_correct:
+            self._battery.last_msg_correct = False
+            _logger.warning(
+                "Battery %d stopped sending data, last timestamp: %s",
+                self._battery.component_id,
+                self._battery.last_msg_timestamp,
+            )
+
+    def _handle_status_inverter_timer(
+        self, _timedelta: timedelta
+    ) -> None:  # Any is probably not the best
+        if self._inverter.last_msg_correct:
+            self._inverter.last_msg_correct = False
+            _logger.warning(
+                "Inverter %d stopped sending data, last timestamp: %s",
+                self._inverter.component_id,
+                self._inverter.last_msg_timestamp,
+            )
+
+    def _check_status(self) -> Optional[Status]:
```

Comment:
I think it is more clear if this remains close to the loop, because we are looking if the status was updated or not, and send it only if it did. This function is not really *checking* the status, it is only checking if it changed or not. It wasn't clear to me what was going on when reading the main loop.

If you prefer to keep it in a separate function, maybe rename it to something more clear, even if it is longer, for example:

```suggestion
    def _get_new_status_if_changed(self) -> Optional[Status]:
```